mmcv/cnn/bricks/non_local.py

```diff
@@ -78,11 +71,38 @@ def __init__(self,
             norm_cfg=norm_cfg,
             act_cfg=None)
 
+        if self.mode != 'gaussian':
+            self.theta = ConvModule(
+                self.in_channels,
+                self.inter_channels,
+                kernel_size=1,
+                conv_cfg=conv_cfg,
+                act_cfg=None)
+            self.phi = ConvModule(
+                self.in_channels,
+                self.inter_channels,
+                kernel_size=1,
+                conv_cfg=conv_cfg,
+                act_cfg=None)
+
+        if self.mode == 'concatenation':
+            self.concat_project = ConvModule(
+                self.inter_channels * 2,
+                1,
+                kernel_size=1,
+                stride=1,
+                padding=0,
+                bias=False,
+                act_cfg=dict(type='ReLU'))
+
         self.init_weights(**kwargs)
 
     def init_weights(self, std=0.01, zeros_init=True):
-        for m in [self.g, self.theta, self.phi]:
-            normal_init(m.conv, std=std)
+        if self.mode != 'gaussian':
+            for m in [self.g, self.theta, self.phi]:
+                normal_init(m.conv, std=std)
+        else:
+            normal_init(self.g.conv, std=std)
         if zeros_init:
```

Comment:
We may also add test for `zero_init`.

Reply:
Add it in `def test_nonlocal()`.

Reply:
Strange. In coverage report 106-112 is not tested. Not sure about the reason.

Reply:
Codes in 106-112 are tested now.